week3/predict_citibike.Rmd

```diff
@@ -0,0 +1,176 @@
+---
+title: "predict_citibike"
+output: html_document
+date: '2022-06-16'
+---
+
+```{r setup, include=FALSE}
+library(tidyverse)
+library(dplyr)
+library(scales)
+library(modelr)
+library(lubridate)
+library(broom)
+
+
+
+theme_set(theme_bw())
+options(repr.plot.width=4, repr.plot.height=3)
+trips_per_day <- read_tsv("trips_per_day.tsv")
+
+#Adding holiday data to trips per day
+holidays <- read.table('holidays', sep = ',', header = FALSE, col.names = c("row", "ymd", "holiday_name")) %>%
+  select(ymd) %>%
+  mutate(ymd = as.Date(ymd), is_holiday = 1)
+
+trips_per_day <- left_join(trips_per_day, holidays)
+trips_per_day[is.na(trips_per_day)] = 0
+
+
+#include weekdays and factorize them
+trips_per_day$weekday <- wday(trips_per_day$ymd, week_start=1)
+trips_per_day$weekday <- as.factor(trips_per_day$weekday)
+
+```
+
+now we set up to split off the validation and the test set.
+
+```{r cars}
+
+set.seed(25)
+
+num_days <- nrow(trips_per_day)
+frac_train <- 0.9
+num_train <- floor(num_days * frac_train)
+
+ndx <- sample(1:num_days, num_train, replace=F)
+trips_per_day_split <- trips_per_day[ndx, ]
+trips_per_day_test <- trips_per_day[-ndx, ]
+
+#now split for a validation set
+
+num_days <- nrow(trips_per_day_split)
+num_train <- floor(num_days * frac_train)
+ndx <- sample(1:num_days, num_train, replace=F)
+trips_per_day_train <- trips_per_day_split[ndx, ]
+trips_per_day_validate <- trips_per_day_split[-ndx, ]
+
+
+```
+
+## Part 3
+
+I'll replicate the process of making multiple linear models and see if i get similar results:
+
+```{r pressure}
+K <- 1:8
+train_err <- c()
+validate_err <- c()
+for (k in K) {
+  
+    # fit on the training data
+    model <- lm(num_trips ~ poly(tmin, k, raw = T), data=trips_per_day_train)
+    
+    # evaluate on the training data
+    train_err[k] <- sqrt(mean((predict(model, trips_per_day_train) - trips_per_day_train$num_trips)^2))
+
+    # evaluate on the validate data
+    validate_err[k] <- sqrt(mean((predict(model, trips_per_day_validate) - trips_per_day_validate$num_trips)^2))
+}
+
+
+plot_data <- data.frame(K, train_err, validate_err) %>%
+  gather("split", "error", -K)
+
+ggplot(plot_data, aes(x=K, y=error, color=split)) +
+  geom_line() +
+  scale_x_continuous(breaks=K) +
+  xlab('Polynomial Degree') +
+  ylab('RMSE')
+
+
+```
+
+Yeah these are pretty similar results, though I'm not sure why the lables are reversed. Now I'm gonna do my thing with the linear model.

```

Comment:
The labels are correct, but their being switched in terms of rank is not a big deal. We are working with a pretty small validation set, so weird things can happen like this.